internal/services/contracts_events_consumer_test.go

```diff
@@ -182,19 +188,210 @@ func TestUpdatedTimestamp(t *testing.T) {
+	}
+
+	cm := common.BytesToAddress([]byte{uint8(9)})
+	autopiUnit := models.AutopiUnit{
+		UserID:       null.StringFrom("SomeID"),
+		OwnerAddress: null.BytesFrom(cm.Bytes()),
+		CreatedAt:    time.Now(),
+		UpdatedAt:    time.Now(),
+		TokenID:      types.NewNullDecimal(new(decimal.Big).SetBigMantScale(big.NewInt(tokenID), 0)),
+	}
+
+	err := autopiUnit.Insert(s.ctx, s.pdb.DBS().Writer, boil.Infer())
+	s.assert.NoError(err)
+
+	c := NewContractsEventsConsumer(s.pdb, &s.logger, s.settings)
+
+	err = c.processMessage(msg)
+	s.assert.EqualError(err, "Handler not provided for contract")
+}
+
+func Test_Ignore_Transfer_Unit_Not_Found(t *testing.T) {
+	s := initCEventsTestHelper(t)
+	defer s.destroy()
+
+	tokenID := int64(4)
+	factoryResp := transferEventsPayloadFactory(1, 3, 5, s.settings.DIMORegistryChainID, AftermarketDeviceContractAddress)
+
+	msg := &message.Message{
+		Payload: []byte(factoryResp.payload),
+	}
+
+	cm := common.BytesToAddress([]byte{uint8(9)})
+	autopiUnit := models.AutopiUnit{
+		UserID:       null.StringFrom("SomeID"),
+		OwnerAddress: null.BytesFrom(cm.Bytes()),
+		CreatedAt:    time.Now(),
+		UpdatedAt:    time.Now(),
+		TokenID:      types.NewNullDecimal(new(decimal.Big).SetBigMantScale(big.NewInt(tokenID), 0)),
 	}
 
+	err := autopiUnit.Insert(s.ctx, s.pdb.DBS().Writer, boil.Infer())
+	s.assert.NoError(err)
+
+	c := NewContractsEventsConsumer(s.pdb, &s.logger, s.settings)
+
+	err = c.processMessage(msg)
+	s.assert.EqualError(err, "record not found as this might be a newly minted device")
+}
+
+func convertTokenIDToDecimal(t string) types.Decimal {
+	ti, ok := new(decimal.Big).SetString(t)
+	if !ok {
+		return types.Decimal{}
+	}
+
+	return types.NewDecimal(ti)
+}
+
+func transferEventsPayloadFactory(from, to int, tokenID int64, dimoChainID int64, contractAddrress string) eventsFactoryResp {
+	contractAddr := common.HexToAddress(contractAddrress)
+	frmAddr := common.BytesToAddress([]byte{uint8(from)})
+	toAddr := common.BytesToAddress([]byte{uint8(to)})
+	tkID := convertTokenIDToDecimal(fmt.Sprint(tokenID))
+
+	payload := fmt.Sprintf(`{
+		"data": {
+			"contract": "%s",
+			"transactionHash": "0x29d1aa4f5eb409bf7d334a7f50fcba50264fbefe00c991cc278f444eb64fdfe5",
+			"eventSignature": "0x61a24679288162b799d80b2bb2b8b0fcdd5c5f53ac19e9246cc190b60196c359",
+			"eventName": "%s",
+			"arguments": {
+				"from": "%s",
+				"to": "%s",
+				"tokenId": %s
+			}
+		},
+		"type": "zone.dimo.contract.event",
+		"source": "chain/%d"
+	}`, contractAddr.String(), "Transfer", frmAddr.String(), toAddr.String(), tkID, dimoChainID)
+
+	return eventsFactoryResp{
+		payload: payload,
+		args: mockTestArgs{
+			contract: contractAddr,
+			tokenID:  tkID,
+			from:     frmAddr,
+			to:       toAddr,
+		},
+	}
+}
+
+// Utility/Helper functions
+
+// Creates a specific number of payloads that the event consumer can parse and process. Can start at an arbitrary point in the array.
+// @Param from - start from index.
```

Comment:
This kind of thing isn't common in Godoc, but I don't think it's harming anyone.